onboarding/source/amb_light_service.c

```diff
@@ -34,20 +44,78 @@ static void lightServiceTask(void * pvParameters);
+    {
+        // Create light service task
+        lightServiceTaskHandle = xTaskCreateStatic( lightServiceTask,             /* Function that implements the task. */
+                                                    LIGHT_SERVICE_NAME,            /* Text name for the task. */
+                                                    LIGHT_SERVICE_STACK_SIZE,      /* Stack size in words, not bytes. */
+                                                    NULL,                       /* Parameter passed into the task. */
+                                                    LIGHT_SERVICE_PRIORITY,        /* Priority at which the task is created. */
+                                                    lightServiceTaskStack,        /* Array to use as the task's stack. */
+                                                    &lightServiceTaskBuffer);     /* Used to hold the task's data structure */
+    }
+
+    if (lightServiceTaskHandle == NULL)
+        return OBC_ERR_CODE_TASK_CREATION_FAILED;
+    
+    if (lightServiceQueueHandle == NULL) 
+    {
+        lightServiceQueueHandle = xQueueCreateStatic ( QUEUE_LENGTH,
+                                                       ITEM_SIZE,
+                                                       lightServiceQueueStorage,
+                                                       &lightServiceQueue);
+    }
+
+    if(lightServiceQueueHandle == NULL)
+        return OBC_ERR_CODE_QUEUE_CREATION_FAILED;
+
+    return OBC_ERR_CODE_SUCCESS;
 
     /* USER CODE END */
 }
 
 static void lightServiceTask(void * pvParameters) {
     /* USER CODE BEGIN */
     // Wait for MEASURE_LIGHT event in the queue and then print the ambient light value to the serial port.
+    if (lightServiceQueueHandle != NULL)
+    {
+        light_event_t event;
+        while (1)
+        {
+            if (xQueueReceive (lightServiceQueueHandle, 
+                                &event,
+                                ( TickType_t) 0) == pdTRUE && event == MEASURE_LIGHT)
+            {
+                adcStartConversion(adcREG1, adcGROUP1);
+                if (adcIsConversionComplete(adcREG1, adcGROUP1) == 8) 
+                {
+                    adcData_t data;
+                    adcGetData(adcREG1, adcGROUP1, &data);
+                    sciPrintf("The light service data id: %lu, data: %u", data.id, data.value);
+                }
+            }
 
+        }
+        
+    }
     /* USER CODE END */
 }
 
 obc_error_code_t sendToLightServiceQueue(light_event_t *event) {
     /* USER CODE BEGIN */
     // Send the event to the queue. Return error code if event was not sent successfully.
+    if (lightServiceQueueHandle != NULL) 
+    {
+        if (xQueueSend( lightServiceQueueHandle,
+                        event,
+                        ( TickType_t ) 0 ) != pdTRUE)
+        {
+            return OBC_ERR_CODE_QUEUE_SEND_FAILED;
+        } 
+        return OBC_ERR_CODE_SUCCESS;
+    }
+    return OBC_ERR_CODE_INVALID_ARG;
```

Comment:
If `lightServiceQueueHandle == NULL`, it'd make more sense to return an error code like `OBC_ERR_CODE_INVALID_STATE` since it'd mean `initLightService` wasn't called successfully. `OBC_ERR_CODE_INVALID_ARG` is used when a function argument is invalid.